fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1026,9 +1032,44 @@ ACTOR Future<Void> commitBatch(
 			trs[t].reply.sendError(transaction_too_old());
 		}
 		else {
-			trs[t].reply.sendError(not_committed());
+			// If enable the option to report conflicting keys from resolvers, we union all conflicting key ranges here and send back through CommitID
+			if (trs[t].transaction.report_conflicting_keys) {
+				Standalone<VectorRef<KeyRangeRef>> conflictingEntries;
```

Comment:
This code is in a critical path - we need to be very careful to make this as CPU efficient as possible. I think we can optimize this a bit more, let's talk in person when you have some time.

Reply:
Pushed this sorting and union process to client-side. Thanks.